carbon/app.py

```diff
@@ -16,83 +16,148 @@
+    "PART-TIME FLEXIBLE/LL",
 )
 
-ENV_VARS = ('FTP_USER', 'FTP_PASS', 'FTP_PATH', 'FTP_HOST', 'FTP_PORT',
-            'CARBON_DB')
+ENV_VARS = (
+    "FTP_USER",
+    "FTP_PASS",
+    "FTP_PATH",
+    "FTP_HOST",
+    "FTP_PORT",
+    "CARBON_DB",
+)
 
 
 def people():
     """A person generator.
 
     Returns an iterator of person dictionaries.
     """
-    sql = select([persons.c.MIT_ID, persons.c.KRB_NAME_UPPERCASE,
-                  persons.c.FIRST_NAME, persons.c.MIDDLE_NAME,
-                  persons.c.LAST_NAME, persons.c.EMAIL_ADDRESS,
-                  persons.c.DATE_TO_FACULTY,
-                  persons.c.ORIGINAL_HIRE_DATE, dlcs.c.DLC_NAME,
-                  persons.c.PERSONNEL_SUBAREA_CODE,
-                  persons.c.APPOINTMENT_END_DATE, orcids.c.ORCID,
-                  dlcs.c.ORG_HIER_SCHOOL_AREA_NAME,
-                  dlcs.c.HR_ORG_LEVEL5_NAME, ]) \
-        .select_from(persons.outerjoin(orcids).join(dlcs)) \
-        .where(persons.c.EMAIL_ADDRESS != None) \
-        .where(persons.c.LAST_NAME != None) \
-        .where(persons.c.KRB_NAME_UPPERCASE != None) \
-        .where(persons.c.KRB_NAME_UPPERCASE != 'UNKNOWN') \
-        .where(persons.c.MIT_ID != None) \
-        .where(persons.c.ORIGINAL_HIRE_DATE != None) \
-        .where(persons.c.APPOINTMENT_END_DATE >= datetime(2009, 1, 1)) \
-        .where(func.upper(dlcs.c.ORG_HIER_SCHOOL_AREA_NAME).in_(AREAS)) \
-        .where(persons.c.PERSONNEL_SUBAREA_CODE.in_(PS_CODES)) \
-        .where(func.upper(persons.c.JOB_TITLE).in_(TITLES))  # noqa: E711
+    sql = (
+        select(
+            persons.c.MIT_ID,
+            persons.c.KRB_NAME_UPPERCASE,
+            persons.c.FIRST_NAME,
+            persons.c.MIDDLE_NAME,
+            persons.c.LAST_NAME,
+            persons.c.EMAIL_ADDRESS,
+            persons.c.DATE_TO_FACULTY,
+            persons.c.ORIGINAL_HIRE_DATE,
+            dlcs.c.DLC_NAME,
+            persons.c.PERSONNEL_SUBAREA_CODE,
+            persons.c.APPOINTMENT_END_DATE,
+            orcids.c.ORCID,
+            dlcs.c.ORG_HIER_SCHOOL_AREA_NAME,
+            dlcs.c.HR_ORG_LEVEL5_NAME,
+        )
+        .select_from(persons)
+        .outerjoin(orcids)
+        .join(dlcs)
+        .where(persons.c.EMAIL_ADDRESS.is_not(None))
+        .where(persons.c.LAST_NAME.is_not(None))
+        .where(persons.c.KRB_NAME_UPPERCASE.is_not(None))
+        .where(persons.c.KRB_NAME_UPPERCASE != "UNKNOWN")
+        .where(persons.c.MIT_ID.is_not(None))
+        .where(persons.c.ORIGINAL_HIRE_DATE.is_not(None))
+        .where(persons.c.APPOINTMENT_END_DATE >= datetime(2009, 1, 1))
+        .where(func.upper(dlcs.c.ORG_HIER_SCHOOL_AREA_NAME).in_(AREAS))
+        .where(persons.c.PERSONNEL_SUBAREA_CODE.in_(PS_CODES))
+        .where(func.upper(persons.c.JOB_TITLE).in_(TITLES))
+    )
     with closing(engine().connect()) as conn:
-        for row in conn.execute(sql):
-            yield dict(zip(row.keys(), row))
+        result = conn.execute(sql)
```

Comment:
I would use the original format of this block:
```
for row in conn.execute(sql):
            yield dict(zip(result.keys(), row))
```
No need to create a `result` variable if it's only used once

Reply:
With the update, I think the `CursorResult` object is only retrieved once rather than recreated for every row. This example code shown below is also indicated in the SQL Alchemy documentation!: https://docs.sqlalchemy.org/en/20/core/connections.html.

<img width="871" alt="image" src="https://github.com/MITLibraries/carbon/assets/29102054/f27e907d-ca5c-48e3-9c09-ff59dc6cffbf">

Let me know what you think of keeping it as-is to make it consistent with recommendation from doc!

Reply:
@ehanson8 and I had a conversation offline and agreed to keep it as-is for now!